File: in_Java/0997 Find the Town Judge.java
class Solution {
    public int findJudge(int N, int[][] trust) {
        int[] indegree = new int[N];
        int[] outdegree = new int[N];
        for(int i = 0; i < trust.length; i++){
            int x = trust[i][0];
            int y = trust[i][1];
            outdegree[x-1]++;
            indegree[y-1]++;
        }
        for(int i = 0; i < N; i++){
            if(indegree[i] == N-1 && outdegree[i] == 0) return i+1;
        }
        return -1;
    }
}

File: in_Python/0997 Find the Town Judge.py
class Solution(object):
    def findJudge(self, N, trust):
        """
        :type N: int
        :type trust: List[List[int]]
        :rtype: int
        """
        if trust == None or len(trust) == 0:
            if N == 1:
                return N
            else:
                return -1
        indeg = []
        outdeg = []
        for i in range(N+1):
            indeg.append(0)
            outdeg.append(0)
        for t in trust:
            a = t[0]
            b = t[1]
            outdeg[a] += 1
            indeg[b] += 1
        #print(indeg)
        #print(outdeg)
        for i in range(N+1):
            if outdeg[i] == 0 and indeg[i] == N-1:
                return i
        return -1
            
        